extensions/interactions/base.py

```diff
@@ -192,23 +211,48 @@ def to_dict(self):
             'instructions': self.instructions,
             'narrow_instructions': self.narrow_instructions,
             'default_outcome_heading': self.default_outcome_heading,
+            'rule_descriptions': self._rule_description_strings,
         }
 
-        # Add information about rule descriptions corresponding to the answer
-        # type for this interaction.
-        result['rule_descriptions'] = (
-            rule_domain.get_description_strings_for_obj_type(
-                self.answer_type))
+    def get_rule_description(self, rule_name):
+        """Gets a rule description, given its name."""
+        if rule_name not in self.rules:
+            raise Exception('Could not find rule with name %s' % rule_name)
+        else:
+            return self.rules[rule_name]['description']
+
+    def get_rule_param_list(self, rule_name):
+        """Gets the parameter list for a given rule."""
+        description = self.get_rule_description(rule_name)
 
-        return result
+        param_list = []
+        while description.find('{{') != -1:
+            opening_index = description.find('{{')
+            description = description[opening_index + 2:]
 
-    def get_rule_by_name(self, rule_name):
-        """Gets a rule given its name."""
-        try:
-            return next(
-                r for r in self.rules if r.__name__ == rule_name)
-        except StopIteration:
-            raise Exception('Could not find rule with name %s' % rule_name)
+            bar_index = description.find('|')
+            param_name = description[: bar_index]
+            description = description[bar_index + 1:]
+
+            closing_index = description.find('}}')
+            normalizer_string = description[: closing_index]
+            description = description[closing_index + 2:]
+
+            param_list.append(
+                (param_name, getattr(objects, normalizer_string))
+            )
+
+        return param_list
+
+    def get_rule_param_type(self, rule_name, rule_param_name):
+        """Gets the parameter type for a given rule parameter name."""
+        rule_param_list = self.get_rule_param_list(rule_name)
+
+        for item in rule_param_list:
```

Comment:
Can this be "for (..., ...) in rule_param_list", with each of the ellipsis replaced by a meaningful noun?